### ISSUE TITLE
`scip stats` fails with "no such file or directory"

### ISSUE THREAD

**Issue by abitrolly** [sourcegraph/scip#98]
Repeatable with latest version.
```
$ scip stats --from - < /data/s/gitlab-development-kit/gdk.scip
2022/10/02 11:51:47 stat /s/gitlab-development-kit: no such file or directory
```

[gdk.zip](https://github.com/sourcegraph/scip/files/9692118/gdk.zip)

**Comment by olafurpg** [comment]
Thank you for reporting! I am able to reproduce with the `gdk.scip` file you've uploaded. The `scip stats` command tries to read the text contents of the referenced documents in the index file and this error is reported because `/s/gitlab-devlopment-kit` does not exist on my computer. The `stats` command should report a more helpful error message and implement more graceful error handling.

**Comment by abitrolly** [comment]
@olafurpg the actual path is `/data/s/gitlab-development-kit`. There was no `/s/gitlab-development-kit` on my computer when `gdk.scip` was generated.

**Comment by matthewnitschke-wk** [comment]
@abitrolly  Was able to replicate and find the problem

If the `metadata` field on the generated scip bindings use a `projectRoot` with a `file://path/to/root`, the `scip stats` command will fail with the reported error

Looking at the `gdk.zip` file you've uploaded, the `Metadata` field looks like this:
```
Metadata: &scip.Metadata{
    Version:  0,
    ToolInfo: &scip.ToolInfo{
      Name:      "scip-ruby",
      Version:   "0.5",
      Arguments: []string{
        "FIXME",
      },
    },
    ProjectRoot:          "file://data/s/gitlab-development-kit",
    TextDocumentEncoding: 1,
  },
```

If `ProjectRoot` is set to: `file:///data/s/gitlab-development-kit`, `scip stats` should result in a successful run

Not sure if `scip-ruby` still outputs the `ProjectRoot` this way, but something to retry/redirect to: https://github.com/sourcegraph/scip-ruby/issues

**Comment by abitrolly** [comment]
@matthewnitschke-wk good catch. I can see this too.

```sh
s ./scip print index.scip 
&scip.Index{
  Metadata: &scip.Metadata{
    Version:  0,
    ToolInfo: &scip.ToolInfo{
      Name:      "scip-ruby",
      Version:   "0.5",
      Arguments: []string{
        "FIXME",
      },
    },
    ProjectRoot:          "file://data/s/gitlab-development-kit",
    TextDocumentEncoding: 1,
  }
...
```

Not sure how to edit the file to validate the fix. Looks like I need `protobuf` editor.

**Comment by abitrolly** [comment]
The file was produced with the command below that was run from `/data/s/gitlab-development-kit` dir.
```sh
../scip-ruby --index-file gdk.scip .
```
Trying [v0.3.8](https://github.com/sourcegraph/scip-ruby/blob/scip-ruby/master/CHANGELOG.md) now.

**Comment by abitrolly** [comment]
Still outputs invalid project root.
```go
&scip.Index{
  Metadata: &scip.Metadata{
    Version:  0,
    ToolInfo: &scip.ToolInfo{
      Name:      "scip-ruby",
      Version:   "0.3.8",
      Arguments: []string{
        "FIXME",
      },
    },
    ProjectRoot:          "file://data/s/gdk",
    TextDocumentEncoding: 1,
  },
...
```
My internal paranoia tells me that storing ProjectRoot is insecure. Why use it at all?

**Comment by olafurpg** [comment]
The `scip snapshot` command has a `--project-root` option that allows you to override the project root directory in cases like this. It would make sense to have the `scip stats` command support a similar flag.

**Comment by abitrolly** [comment]
`scip-ruby` has some hacky comments about `ProjectRoot` https://github.com/sourcegraph/scip-ruby/pull/27/files

I am trying to port `--project-root` to `scip stats`.

**Comment by abitrolly** [comment]
@olafurpg looks like `scip snapshot` fails on this too, even before `--project-root` is processed.

**Comment by abitrolly** [comment]
@olafurpg replicated `--project-root` for `scip stats`, but tests fail with "unexpected HTTP status code received from server". Please take a look at https://github.com/sourcegraph/scip/pull/187

**Comment by olafurpg** [comment]
Closing this as fixed by #187. It's expected that `scip stats` fails if it's unable to resolve the absolute path of a document in the provided SCIP index. You can override the project root with the new `--project-root` flag. Thank you @abitrolly for the contribution!

I'm open to adding some kind of `--failure-mode=lenient` option to ignore files that don't resolve to absolute paths, but this can be implemented separately from this issue.

**Comment by abitrolly** [comment]
`--project-root` logic falls back to current dir if absolute path from SCIP can not be found, so it should be good now.
